### PR TITLE
refactor(generator): separate out longrunning functions

### PR DESCRIPTION
**Pull request by scotthart**
part of the work for #10191

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11546)

### CODE REVIEW

File: generator/internal/longrunning_test.cc [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+  google::protobuf::SimpleDescriptorDatabase simple_db_;
+  google::protobuf::compiler::SourceTreeDescriptorDatabase source_tree_db_;
+  google::protobuf::MergedDescriptorDatabase merged_db_;
+
+ protected:
+  DescriptorPool pool_;
+};
+
+TEST_F(LongrunningMethodVarsTest, FilesParseSuccessfully) {
+  EXPECT_NE(nullptr, pool_.FindFileByName("google/api/client.proto"));
```
Comment (dbolduc) [inline generator/internal/longrunning_test.cc:333]: nit: `EXPECT_THAT(foo, Not(IsNull()));`

File: generator/internal/longrunning_test.cc [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+
+TEST_F(LongrunningMethodVarsTest,
+       SetLongrunningOperationMethodVarsResponseAndMetadata) {
+  FileDescriptor const* service_file_descriptor =
+      pool_.FindFileByName("google/foo/v1/service.proto");
+  MethodDescriptor const* method =
+      service_file_descriptor->service(0)->method(0);
+  VarsDictionary vars;
+  SetLongrunningOperationMethodVars(*method, vars);
+  EXPECT_THAT(vars, ::testing::Contains(
```
Comment (dbolduc) [inline generator/internal/longrunning_test.cc:349]: `using ::testing::Foo;` so the expectations are closer to english

**Review by dbolduc — APPROVED** [review]

File: generator/internal/longrunning_test.cc (hunk 1 above)
Comment (dbolduc) [reply generator/internal/longrunning_test.cc:333]: Oops, @coryan reminds me that `::testing::NotNull` is a thing

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11546)* status: 0 of 30 files reviewed, 3 unresolved discussions (waiting on @coryan and @dbolduc)

___
*[`generator/internal/longrunning.cc` line 22 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11546#gh-1189101560:-NV1Pdou42PDDNbbKFqe:b-cbwfn3) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/9e841de2f47213eb8dcf112428ecb00c3e63eab1/generator/internal/longrunning.cc#L22)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

hm, we are not consistent about `using` this. I would suggest removing the declaration and spelling it out everywhere.
</blockquote></details>

I was inconsistent in this file, so I opted to make things more consistent and added using declarations for oft used types.
___
*[`generator/internal/longrunning_test.cc` line 333 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11546#gh-1189134685:-NV1Q8KD4diZeGpg50DS:b1xqbqe) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/9e841de2f47213eb8dcf112428ecb00c3e63eab1/generator/internal/longrunning_test.cc#L333)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

Oops, @coryan reminds me that `::testing::NotNull` is a thing
</blockquote></details>

Updated and re-updated to use NotNull.
___
*[`generator/internal/longrunning_test.cc` line 349 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11546#gh-1189134975:-NV1PtLnDeFP_98J_lxK:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/9e841de2f47213eb8dcf112428ecb00c3e63eab1/generator/internal/longrunning_test.cc#L349)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

`using ::testing::Foo;` so the expectations are closer to english
</blockquote></details>

Done.

**Pull request merged**